### PR TITLE
[#165882756] Update and uniform search bar component style

### PR DESCRIPTION
**Pull request by laurapugliese**
For the moment extract the search in component. Inserted search component in the header global component and used it in the messages list

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#165882756](https://www.pivotaltracker.com/story/show/165882756): Come CIT voglio un comportamento omogeneo della search bar fra i vari screen

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by cloudify** [comment]
This pull request **introduces 1 alert** when merging f7d15eaf6442bf5cc40d5a85502b79c1a13c74bd into d3d20dd19e73b7bcc48a6e017b656cb71c0067e6 - [view on LGTM.com](https://lgtm.com/projects/g/teamdigitale/io-app/rev/pr-ff5ae51344812a3509132e27b82012f96161cfec)

**new alerts:**

* 1 for Unused or undefined state property

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by laurapugliese** [comment]
This PR aims to standardize the search in all the screens in which it is present (message list and service list).

**on Android:**
search_android

**on iOS:**
search_ios_optimized

**Comment by matteodesanti** [comment]
Great job @laurapugliese , OK for me.

File: ts/components/search/SearchButton.tsx [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+      searchText: some(text)
+    });
+    this.updateDebouncedSearchText(text);
+  };
+
+  private updateDebouncedSearchText = debounce((text: string) => {
+    this.setState({
+      debouncedSearchText: some(text)
+    });
+    this.props.dispatchSearchText(this.state.debouncedSearchText);
```
Comment (fpersico) [inline ts/components/search/SearchButton.tsx]: The state is updated async. There is no guarantee that this.state.debouncedSearchText is already updated here. Use the setState callback.

File: ts/components/search/SearchButton.tsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  }, 300);
+
+  private onSearchDisable = () => {
+    this.setState({
+      searchText: none,
+      debouncedSearchText: none
+    });
+    this.props.dispatchResetSearch();
```
Comment (fpersico) [inline ts/components/search/SearchButton.tsx]: dispatchDisableSearch?

**Comment by fpersico** [comment]
@laurapugliese look at the gif. There are 2 bugs:

1) The duplicated IO logo
2) On screen change from ServiceList to MessageList the searchbar is rendered uncorrectly

Bug

**Comment by laurapugliese** [comment]
> @laurapugliese look at the gif. There are 2 bugs:
> 
> 1. The duplicated IO logo
> 2. On screen change from ServiceList to MessageList the searchbar is rendered uncorrectly
> 
> Bug

@fpersico This is a problem I had during the merge with the master before, when I still had the PR in Draft (after changes were made to the header while I was working on the branch). In the last commit 3379440 this problem should not be there. I tried again now locally to start the branch and the bug does not match it. Which device did you test on?

**Comment by fpersico** [comment]
@laurapugliese my fault, i was not on the last commit. I am going to check it again.

File: ts/components/search/SearchButton.tsx (hunk 2 above)
Comment (laurapugliese) [reply ts/components/search/SearchButton.tsx]: Renamed it in the commit 285ad93

File: ts/components/search/SearchButton.tsx (hunk 1 above)
Comment (laurapugliese) [reply ts/components/search/SearchButton.tsx]: Fixed it in the commit 285ad93

**Pull request merged**